app/src/main/java/uk/gov/onelogin/login/LoginGraphObject.kt

```diff
@@ -0,0 +1,99 @@
+package uk.gov.onelogin.login
+
+import androidx.activity.compose.BackHandler
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.composable
+import androidx.navigation.navigation
+import uk.gov.onelogin.login.ui.LoadingScreen
+import uk.gov.onelogin.login.ui.PasscodeInfoScreen
+import uk.gov.onelogin.login.ui.SignInErrorScreen
+import uk.gov.onelogin.login.ui.biooptin.BiometricsOptInScreen
+import uk.gov.onelogin.login.ui.splash.SplashScreen
+import uk.gov.onelogin.login.ui.welcome.WelcomeScreen
+import uk.gov.onelogin.mainnav.MainNavRoutes
+import uk.gov.onelogin.navigation.hasPreviousBackStack
+import uk.gov.onelogin.navigation.navigateToOfflineErrorScreen
+import uk.gov.onelogin.ui.error.ErrorRoutes.OFFLINE_ERROR_TRY_AGAIN_KEY
+
+object LoginGraphObject {
+    @Suppress("LongMethod")
+    fun NavGraphBuilder.loginGraph(
+        navController: NavHostController
+    ) {
+        navigation(
+            route = LoginRoutes.Root.getRoute(),
+            startDestination = LoginRoutes.Start.getRoute()
+        ) {
+            composable(
```

Comment:
I wonder if it is best to split each navigation block composable into its own function to reduce the length of this function and make it easier to read. Something to look into at a later date maybe.

Reply:
Yeah agreed, definitely could be a nice idea. Will probably not include that here though, bit overloaded already in this PR